src/pages/DistanceCalculator/hooks.ts

```diff
@@ -74,11 +75,34 @@ function handleNauticalMile(
 
 export function useDistanceCalculator(): DistanceCalculatorHookOutput {
   const [kilometerInput, setKilometerInput] = useState<string>("");
+  const {previous: prevKilometerInput, updatePrevious: updatePrevKilometerInput} = usePrevious<string>(kilometerInput);
+
   const [mileInput, setMileInput] = useState<string>("");
+  const {previous: prevMileInput, updatePrevious: updatePrevMileInput} = usePrevious<string>(mileInput);
+
   const [nauticalMileInput, setNauticalMileInput] = useState<string>("");
+  const {previous: prevNauticalMileInput, updatePrevious: updatePrevNauticalMileInput} = usePrevious<string>(nauticalMileInput);
+
   const [selection, setSelection] = useState<DistanceUnit>(DistanceUnit.Kilometer);
+  const {previous: prevSelection, updatePrevious: updatePrevSelection} = usePrevious<DistanceUnit>(selection);
```

Comment:
@iamdejan I think we can use `useMemo` and/or `useCallback` to avoid if-else, but with major code restructure.

Reply:
update: I cannot complete remove if-else, but at least this one is more readable.